src/spawn_sync.cc

```diff
@@ -502,10 +501,11 @@ Maybe<bool> SyncProcessRunner::TryInitializeAndRunLoop(Local<Value> options) {
   }
   uv_process_.data = this;
 
-  for (uint32_t i = 0; i < stdio_count_; i++) {
-    SyncProcessStdioPipe* h = stdio_pipes_[i].get();
-    if (h != nullptr) {
-      r = h->Start();
+  std::vector<std::unique_ptr<SyncProcessStdioPipe>>::iterator iter;
+  for (iter = stdio_pipes_.begin(); iter != stdio_pipes_.end(); ++iter) {
```

Comment:
Feel free to use a `for (const auto& pipe : stdio_pipes_)` loop here :)

Reply:
I'm going to explicitly request this change.

Reply:
FTR here are some of the guidelines I'm basing my request on:
[ES.55](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-range-checking): Avoid the need for range checking
[ES.6](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-cond): Declare names in for-statement initializers and conditions to limit scope
[ES.11](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto): Use auto to avoid redundant repetition of type names
[ES.20](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-always): Always initialize an object